src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusTopicResource.cs

```diff
@@ -39,7 +39,10 @@ public string TopicName
     /// <summary>
     /// Gets the connection string expression for the Azure Service Bus Topic.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression => Parent.ConnectionStringExpression;
+    public ReferenceExpression ConnectionStringExpression =>
+        Parent.IsEmulator
+        ? ReferenceExpression.Create($"{Parent.ConnectionStringExpression}EntityPath={TopicName}")
+        : ReferenceExpression.Create($"Endpoint={Parent.ConnectionStringExpression};EntityPath={TopicName}");
```

Comment:
Can we follow the same pattern as

https://github.com/dotnet/aspire/blob/08b424e1fa0b28aa225a67e2c6a7b5b959037f05/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsResource.cs#L50-L82